src/web/components/Core/KeySecretReveal.tsx

```diff
@@ -0,0 +1,113 @@
+import { useEffect, useState } from 'react';
+
+import Popover from './Popover';
+import { StatusPopup } from './StatusPopup';
+
+import './KeySecretReveal.scss';
+
+const MAX_SHOWN_VALUE_LENGTH = 20;
+
+function ViewKeyButton({
```

Comment:
maybe `ViewSecretButton` would be a better name here? I think references to `Key` in this file are a bit confusing